activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter/src/test/java/org/activiti/starter/tests/cmdendpoint/CommandEndpointIT.java

```diff
@@ -0,0 +1,421 @@
+    public static final String PROCESS_INSTANCES_RELATIVE_URL = "/v1/process-instances/";
+    private static final String TASKS_URL = "/v1/tasks/";
+
+    private static final String SIMPLE_PROCESS = "SimpleProcess";
+
+    private static AtomicBoolean startedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean suspendedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean activatedProcessInstanceAck = new AtomicBoolean(false);
+    private static AtomicBoolean claimedTaskAck = new AtomicBoolean(false);
+    private static AtomicBoolean releasedTaskAck = new AtomicBoolean(false);
+    private static AtomicBoolean completedTaskAck = new AtomicBoolean(false);
+
+
+
+    private StartProcessInstanceCmd startProcessInstanceCmd;
+
+    private SuspendProcessInstanceCmd suspendProcessInstanceCmd;
+
+    private static String testProcessInstanceId;
+
+    @Before
+    public void setUp() throws Exception {
+        keycloaktestuser = "hruser";
+        //don't need to set password as same password as testuser
+        accessToken = authenticateUser();
+
+        // Get Available Process Definitions
+        ResponseEntity<PagedResources<ProcessDefinition>> processDefinitions = getProcessDefinitions();
+        assertThat(processDefinitions.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        assertThat(processDefinitions.getBody().getContent()).hasSize(6); //if a new definition is added then this is expected to be increased
+        for (ProcessDefinition pd : processDefinitions.getBody().getContent()) {
+            processDefinitionIds.put(pd.getName(),
+                                     pd.getId());
+        }
+    }
+
+    @EnableAutoConfiguration
+    public static class StreamHandler {
+
+        @StreamListener(MessageClientStream.MY_CMD_RESULTS)
+        public void consumeResults(CommandResults results) {
+            assertThat(results).isNotNull();
+            if (results instanceof StartProcessInstanceResults) {
+                assertThat(((StartProcessInstanceResults) results).getProcessInstanceId()).isNotEmpty();
+                testProcessInstanceId = ((StartProcessInstanceResults) results).getProcessInstanceId();
+                startedProcessInstanceAck.set(true);
+            } else if (results instanceof SuspendProcessInstanceResults) {
+                suspendedProcessInstanceAck.set(true);
+            } else if (results instanceof ActivateProcessInstanceResults) {
+                activatedProcessInstanceAck.set(true);
+            } else if (results instanceof ClaimTaskResults) {
+                claimedTaskAck.set(true);
+            } else if (results instanceof ReleaseTaskResults) {
+                releasedTaskAck.set(true);
+            } else if (results instanceof CompleteTaskResults) {
+                completedTaskAck.set(true);
+            }
+        }
+    }
+
+    @Test
+    public void eventBasedStartProcessTests() throws Exception {
```

Comment:
I think that this test could be split in several ones: one for command.
* startProcessInstanceCmdShouldCreateANewProcessInstance
* suspendProcessInstanceCmdShouldPutTheProcessInTheSuspendedState
* activateProcessShouldPutTheProcessBackToActiveState
...